### ISSUE TITLE
Improve error for invalid URLs

### ISSUE THREAD

**Issue by Dante-dan** [nitrojs/nitro#717]
### Environment

nitropack: 1.0.0
nuxt3: 1.0.0
node: node18

### Reproduction

nuxt3

```
pnpm build
```

```
pnpm preview
```

### Describe the bug

invalid url will cause 500 error, because `new URL()`  . It shoud be  `catch` exception and throw not match 404.

sample URl : `http://localhost:3000/.%AE/WEB-INF/esp_ldap_context.property`

https://github.com/unjs/nitro/blob/8e06f2e4a65c6de0d2f2f3a5ef156ae3821ed60a/src/runtime/static.ts#L18~L22

``` javascript

// runtime/static.ts
  let id = decodeURIComponent(
    withLeadingSlash(
      withoutTrailingSlash(parseURL(event.node.req.url).pathname)
    )
  );

// [nuxt] [request error] [unhandled] [500] URI malformed
```

**Comment by cjpearson** [comment]
I believe this has possibly been resolved with https://github.com/unjs/nitro/pull/1459. By using `decodePath` from `ufo` instead of `decodeURIComponent` errors with invalid URIs will be caught and ignored. 

If this is the expected behavior rather than throwing a 404, I think this can be closed.

**Comment by cjpearson** [comment]
@pi0 I think after #1459 and #2657 this can be closed.

**Comment by cjpearson** [comment]
@pi0 I think this can be closed with the fixes in #1459 and  #2658
